src/comparator/field_comparator.py

```diff
@@ -106,14 +106,8 @@ def compare(self):
         ):
             # Version update is allowed here, for example from `.example.v1.Enum` to `.example.v1beta1.Enum`.
             # But from `.example.v1.Enum` to `.example.v2.EnumUpdate` is breaking.
-            api_version_original = self.field_original.api_version
-            api_version_update = self.field_update.api_version
-            transformed_type_name = (
-                self.field_original.type_name.value.replace(
-                    api_version_original, api_version_update
-                )
-                if api_version_original
-                else None
+            transformed_type_name = self._transformed_type_name(
+                self.field_original.type_name.value
             )
```

Comment:
This is wrapped in a method (`_transformed_type_name`) below, so that it can be called easily for multiple times, we also need it for map type comparison.